src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Formatter/Utf8Formatter.Integer.Unsigned.X.cs

```diff
@@ -34,12 +33,12 @@ private static bool TryFormatUInt64X(ulong value, byte precision, bool useLower,
             // directly with nibbles. There may be opportunity for improvement by special-
             // casing output lengths of 2, 4, 8, and 16 and running them down optimized
             // code paths.
-
             while ((uint)(--computedOutputLength) < (uint)destination.Length)
             {
-                destination[computedOutputLength] = (byte)hexTable[(int)value & 0xf];
+                destination[computedOutputLength] = (byte)(useLower ? HexConverter.ToCharLower((int)value) : HexConverter.ToCharUpper((int)value));
```

Comment:
Evaluating `useLower` every time through the loop is likely going to regress the formatter performance.

Reply:
The commend above says this is not perf sensitive. I changed the code anyway but it could switch to Span version if we need perf improvements here